handler/src/main/java/io/netty/handler/ssl/OpenSslEngine.java

```diff
@@ -1076,21 +1076,31 @@ public void setEnabledProtocols(String[] protocols) {
                 // Enable all and then disable what we not want
                 SSL.setOptions(ssl, SSL.SSL_OP_ALL);
 
+                int opts = SSL.SSL_OP_NO_SSLv2 | SSL.SSL_OP_NO_SSLv3 | SSL.SSL_OP_NO_TLSv1 |
+                   SSL.SSL_OP_NO_TLSv1_1 | SSL.SSL_OP_NO_TLSv1_2;
+
+                // Clear out options which disable protocols
+                SSL.clearOptions(ssl, opts);
```

Comment:
should we clear options before `SSL.setOptions(ssl, SSL.SSL_OP_ALL);`?  We may have to re-visit this in general, but not sure it is in the scope of this PR.